### PR TITLE
Add 7.14 branch 

### PR DESCRIPTION
**Pull request by jmikell821**
- Adds 7.14 to include/exclude lists in central config file
- Adds shared attributes file for 7.14
- Changes 7.x shared attributes to reference 7.15

### CODE REVIEW

File: shared/versions/stack/7.14.asciidoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+:minor-version:          7.14
+:major-version:          7.x
+:prev-major-version:     6.x
+:major-version-only:     7
+:ecs_version:            1.8
+
+//////////
+release-state can be: released | prerelease | unreleased
+//////////
+:release-state:          released
```
Comment (lcawl) [inline shared/versions/stack/7.14.asciidoc:20]: I think this is required until GA:
```suggestion
:release-state:          prerelease
```

File: shared/versions/stack/7.14.asciidoc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+//////////
+is-current-version can be: true | false
+//////////
+:is-current-version:    true
```
Comment (lcawl) [inline shared/versions/stack/7.14.asciidoc:25]: Likewise this changes at GA:
```suggestion
:is-current-version:    false
```

**Review by lcawl — APPROVED** [review]
LGTM

**Comment by lcawl** [comment]
I think an additional change might also be required in https://github.com/elastic/docs/blob/b58d516b34487b13eae1635397cbca67b119cf09/conf.yaml#L928 when the 7.14 branch exists in the https://github.com/elastic/enterprise-search-pubs repo

**Comment by gtback** [comment]
This was failing because the APM Server repo didn't have a 7.14 branch. The branch has since been created, so... 

@elasticmachine run elasticsearch-ci/docs

**Comment by gtback** [comment]
It looks like the failure now is coming from some of the ES build files that were changed in #2127 and #2131. @breskeby would you be able to look and see which branches should be included in the `exclude_branches` lists for the various files? This is the current error:

```
09:15:50 INFO:build_docs:Error executing: git archive --format=tar -o /tmp/docsbuild/CQq8n0FCni/elasticsearch/.temp_git_archive.tar 7.14 buildSrc in GIT_DIR /docs_build/.repos/elasticsearch.git
09:15:50 INFO:build_docs:---out---
09:15:50 INFO:build_docs:
09:15:50 INFO:build_docs:---err---
09:15:50 INFO:build_docs:fatal: pathspec 'buildSrc' did not match any files
09:15:50 INFO:build_docs:
09:15:50 INFO:build_docs:---------
```

File: conf.yaml [hunk 3]
```diff
@@ -252,7 +252,7 @@ contents:
               -
                 repo:   elasticsearch
                 path:   build-tools-internal/
-                exclude_branches:   [ 7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.5, 6.4, 6.3, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
+                exclude_branches:   [ 7.14, 7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.5, 6.4, 6.3, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
```
Comment (breskeby) [inline conf.yaml:255]: this is wrong here. Starting from 7.14 you only need to exclude 7.14 from the path `buildSrc` not from the build-tools-internal` nor `build-tools` path

**Comment by breskeby** [comment]
@jmikell821 I've added a comment as part of a review what you need to fix in the excludes to get this properly working

**Review by breskeby — CHANGES_REQUESTED** [review]

**Comment by gtback** [comment]
Thanks, @breskeby !

File: conf.yaml [hunk 4]
```diff
@@ -248,11 +248,11 @@ contents:
               -
                 repo:   elasticsearch
                 path:   buildSrc/
-                exclude_branches:   [ master, 7.x, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
+                exclude_branches:   [ master, 7.x, 7.14, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
               -
                 repo:   elasticsearch
                 path:   build-tools-internal/
-                exclude_branches:   [ 7.14, 7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.5, 6.4, 6.3, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
+                exclude_branches:   [7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.5, 6.4, 6.3, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
```
Comment (jmikell821) [inline conf.yaml]: @breskeby please let me know if these two changes are correct.

Comment (jmikell821) [reply conf.yaml]: OK the build still failed, so I must have missed something?

Comment (gtback) [reply conf.yaml]: Based on the error, I think you just need to make the same change under the "Plugins and Integrations" book (around line 385).

**Comment by lcawl** [comment]
@chriscressman The latest build errors look like the only remaining piece is the creation of the 7.14 branch in https://github.com/elastic/enterprise-search-pubs

**Comment by lcawl** [comment]
@elasticmachine run elasticsearch-ci/docs rebuild

**Review by breskeby — APPROVED** [review]

File: conf.yaml [hunk 5]
```diff
@@ -383,10 +383,14 @@ contents:
                 repo:   elasticsearch
                 path:   buildSrc/src/main/resources/
                 exclude_branches:   [ master, 7.x, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
```
Comment (breskeby) [inline conf.yaml:385]: I think 7.14 must be added here

File: conf.yaml [hunk 6]
```diff
@@ -383,10 +383,14 @@ contents:
...
+              -
+                repo:   elasticsearch
+                path:   build-tools-internal/src/main/resources/
+                exclude_branches:   [7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.5, 6.4, 6.3, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
               -
                 repo:   elasticsearch
                 path:   build-tools/src/main/resources/
-                exclude_branches:   [ 7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.5, 6.4, 6.3, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
+                exclude_branches:   [ 7.14, 7.13, 7.12, 7.11, 7.10, 7.9, 7.8, 7.7, 7.6, 7.5, 7.4, 7.3, 7.2, 7.1, 7.0, 6.8, 6.7, 6.6, 6.5, 6.4, 6.3, 6.2, 6.1, 6.0, 5.6, 5.5, 5.4, 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0, 1.7, 1.6, 1.5, 1.4, 1.3, 0.90 ]
```
Comment (breskeby) [inline conf.yaml:393]: I think 7.14 must be removed here.

File: conf.yaml (hunk 5 above)
Comment (breskeby) [inline conf.yaml:385]: 7.14 needs to be added here

**Review by breskeby — CHANGES_REQUESTED** [review]
to get the docs build passing some tweaks need to be done

File: conf.yaml (hunk 6 above)
Comment (lcawl) [reply conf.yaml]: Thanks!  I made this change in 275c1f1fd451b5b340777452787c5e1e804fcebb

File: conf.yaml (hunk 5 above)
Comment (lcawl) [reply conf.yaml]: Thanks!  Addressed in 275c1f1fd451b5b340777452787c5e1e804fcebb

**Review by breskeby — APPROVED** [review]
That should make the docs build pass now 👍

**Pull request merged**